Microsoft.Toolkit.Uwp.UI.Controls/ClassicMenu/ClassicMenu.Events.cs

```diff
@@ -0,0 +1,178 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System.Linq;
+using System.Text;
+using Windows.System;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Classic Menu Control defines a menu of choices for users to invoke.
+    /// </summary>
+    public partial class ClassicMenu
+    {
+        private static bool NavigateThroughMenuHeader(KeyEventArgs args, ClassicMenu menu, ClassicMenuItem menuItem, Orientation orientation)
+        {
+            if (orientation == Orientation.Horizontal)
+            {
+                if (args.VirtualKey == VirtualKey.Down)
+                {
+                    menuItem.ShowMenu();
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Left)
+                {
+                    return MoveFocusBackward(menu, menuItem);
+                }
+
+                if (args.VirtualKey == VirtualKey.Right)
+                {
+                    return MoveFocusForward(menu, menuItem);
+                }
+            }
+            else
+            {
+                if (args.VirtualKey == VirtualKey.Right)
+                {
+                    menuItem.ShowMenu();
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Up)
+                {
+                    return MoveFocusBackward(menu, menuItem);
+                }
+
+                if (args.VirtualKey == VirtualKey.Down)
+                {
+                    return MoveFocusForward(menu, menuItem);
+                }
+            }
+
+            return false;
+        }
+
+        private static bool MoveFocusForward(ClassicMenu menu, ClassicMenuItem menuItem)
```

Comment:
because maybe the pressed character is not an arrow. I need to validate the directions only else check if it is a command